rectools/visuals/visual_app.py

```diff
@@ -49,59 +61,91 @@ def __init__(
         self.grouped_interactions = self._group_interactions(
             interactions=self.interactions,
             selected_requests=self.selected_requests,
-            request_colname=self.request_colname,
+            id_col=self.id_col,
             item_data=self.item_data,
         )
         self.grouped_recos = self._group_recos(
             recos=self.recos,
             selected_requests=self.selected_requests,
-            request_colname=self.request_colname,
+            id_col=self.id_col,
             item_data=self.item_data,
         )
 
+    @property
+    def request_names(self) -> tp.List[tp.Hashable]:
+        return list(self.selected_requests.keys())
+
+    @property
+    def model_names(self) -> tp.List[tp.Hashable]:
+        return list(self.grouped_recos.keys())
+
+    @classmethod
+    def _fill_requests_with_random(
+        cls,
+        selected_requests: tp.Dict[tp.Hashable, tp.Hashable],
+        n_random_requests: int,
+        id_col: str,
+        recos: TablesDict,
+    ) -> tp.Dict[tp.Hashable, tp.Hashable]:
+
+        # Leave only those ids that were not predefined by user
+        # Request ids (e.g. user ids) are stored as values in `selected_requests`
+        all_ids = [model_recos[id_col].unique() for model_recos in recos.values()]
+        unique_ids = np.unique(np.hstack(all_ids))
```

Comment:
usually it's faster to use `pd.unique`
<img width="556" alt="image" src="https://github.com/MobileTeleSystems/RecTools/assets/14802567/0caea63d-6b05-402e-9e54-15bc72912634">

Reply:
something pandas is faster then something numpy, now that's a finding :)
fixed

Reply:
haha

pandas is using a hash table here and numpy is using sorting, so unless you need sorted unique values, pandas is faster by design